src/Tus/Client.php

```diff
@@ -48,6 +48,10 @@ class Client extends AbstractTus
      */
     public function __construct(string $baseUri, array $options = [])
     {
+        $options['headers'] = [
+            'Tus-Resumable' => self::TUS_PROTOCOL_VERSION,
+        ] + ($options['headers'] ?? []);
+
```

Comment:
Since we are now implementing the `Tus-Resumable` it now requires server to send `412 Precondition Failed` response whenever server is unable to handle the request. I don't think it's implemented yet.